Jetsnack/app/src/main/java/com/example/jetsnack/model/SnackbarManager.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.jetsnack.model
+
+import androidx.annotation.StringRes
+import java.util.UUID
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+
+data class Message(val id: Long, @StringRes val messageId: Int)
```

Comment:
This will not allow the consumer to pass dynamic strings, or buttons with callbacks. That's fine for this sample, but we might want to think about a more generic approach.

Reply:
I think it's fine to leave it as is for now as we don't have more complex messages to show.